Data access api

anamanica opened the pull request:
# Change Summary

## Overview
Changing the Data Access API documentation to include new features, note important information, and overall just make things more clear for users.

## New Dependencies
None

## New Files
None

## Deleted Files
None

## Updated Files
<!--List each updated file and add bullets to describe the purpose/function of the updates.
This should be high level, but enough detail for the reviewer to understand what purpose(s) the
updated code in the file is serving-->
- index.rst
   - Added several notes into the end of the upload, download, and intro section of the docs.
- openapi.yaml
   - Added new query options to the query section of the docs

## Testing
None

closes #648

anamanica committed:
Doc changes

anamanica committed:
Merge branch 'dev' of github.com:IMAP-Science-Operations-Center/imap_processing into data-access-api

Updating branch

anamanica committed:
Updating

anamanica committed:
Fixing

anamanica committed:
Fixing look

anamanica committed:
Playing with format

anamanica committed:
Fixing again

anamanica committed:
Fixing again

anamanica committed:
Fixing again

anamanica committed:
Fixing again

anamanica committed:
Fixing again

anamanica committed:
Fixing again

daralynnrhode commented on docs/source/data-access-api/index.rst:
```diff
@@ -64,6 +78,8 @@ The API can be accessed from the following URL: https://api.dev.imap-mission.com
    :group:
    :include: /query
 
+*Note:* ``filename``, ``N days ago`` *, and* ``today`` *are all [WIP].*
```
I think for clarity sake it may be better to note these as WIP in their _descriptions_ (or tag it onto the  _name_ ) in `openapi.yml`.

daralynnrhode reviewed (commented):
Besides the one spot, It all looks good to me!

anamanica committed:
Updating Branch
Merge branch 'dev' of github.com:IMAP-Science-Operations-Center/imap_processing into data-access-api

anamanica committed:
Small change

anamanica committed:
More formatting

bourque commented on docs/source/data-access-api/index.rst:
```diff
@@ -11,14 +11,21 @@ The SDC provides a REST API that allows users to upload and download files, as
+The API can be accessed from the following URL [WIP]: https://api.dev.imap-mission.com
 
+*Note: Several sections and links begin with* [WIP]. *As development on the API is ongoing, this indicates
+that the full implementation of the functionality is yet to be completed.*
```
```suggestion
*Note: Several sections and links begin with* [WIP]. *As development on the API is ongoing, this indicates
that the full implementation of the functionality is yet to be completed.*

The API can be accessed from the following URL [WIP]: https://api.dev.imap-mission.com
```

I like that you added this! I would prefer that the note go before the first use of `[WIP]` so the reader is aware of what it means before they first encounter it, but its up to you!

bourque commented on docs/source/data-access-api/openapi.yml:
```diff
@@ -143,7 +144,8 @@ paths:
         - in: query
           name: version
-          description: The version of data product in the format ``vXX-YY`` (e.g. ``v01-01``).
+          description: The version of data product in the format ``vXX-YY`` (e.g. ``v01-01``). You can also choose to
```
```suggestion
          description: The version of data product in the format ``vNNN`` (e.g. ``v001``). You can also choose to
```
This documentation has become outdated and still references the old version number format. Could you update this?

bourque commented on docs/source/data-access-api/openapi.yml:
```diff
@@ -153,6 +155,24 @@ paths:
           required: false
           schema:
             type: string
+        - in: query
+          name: filename
+          description: WIP -- The full filename of the desired file (e.g. ``imap_mag_l1a_burst_20240105_20240105_v01-01.pkts``).
+          required: false
+          schema:
+            type: string
+        - in: query
+          name: N_days_ago
+          description: WIP -- Returns all files acquired in the last 'N' days (with 'N' representing a numeric value).
+          required: false
+          schema:
+            type: string
+        - in: query
+          name: today
+          description: WIP -- Returns all files acquired in the last 24 hours.
+          required: false
+          schema:
+            type: string
```
I would prefer to add these in as we implement these features so we get in the habit of updating the docs in the same PRs that implement said features. Also, that would avoid a situation in which the docs say 'we are going to add this feature' but we don't actually get around to implementing it for several months, which could be a bad look.

anamanica replied:
Totally agree with you!

bourque reviewed (changes requested):
Thanks for making these updates! I have a few small suggestions, otherwise this looks good!

This can be in a future PR, but I would like to see these docs contain all of the info that is currently provided in the `imap-data-access` repo's `README` file. I created an issue for this: https://github.com/IMAP-Science-Operations-Center/imap-data-access/issues/75.  In that way, all of the documentation for the IMAP Data Access API will be in one place, instead of spread between the two.

anamanica committed:
Updating branch

Merge branch 'dev' of github.com:IMAP-Science-Operations-Center/imap_processing into data-access-api

anamanica committed:
PR updates

anamanica committed:
I changed my mind on format

bourque reviewed (approved):
Thanks for updating, looks good!
